health-integrations/src/main/java/com/netflix/runtime/health/eureka/EurekaHealthStatusBridgeModule.java

```diff
@@ -41,38 +51,37 @@ protected void configure() {
        bind(ApplicationEventListener.class).to(EurekaHealthStatusInformingApplicationEventListener.class).asEagerSingleton();
     }
 
-    private static class EurekaHealthStatusInformingApplicationEventListener implements ApplicationEventListener<HealthCheckStatusChangedEvent> {
-    
-        private final ApplicationEventDispatcher eventDispatcher;
-        private final ApplicationInfoManager applicationInfoManager;
-        private final HealthCheckAggregator healthCheckAggregator;
-        private final EurekaClient eurekaClient;
+    private static class EurekaHealthStatusInformingApplicationEventListener
+            implements ApplicationEventListener<HealthCheckStatusChangedEvent>, LifecycleListener {
+
+        @com.google.inject.Inject(optional=true)
+        private ApplicationEventDispatcher eventDispatcher;
```

Comment:
Change these to Providers? Per documentation for LifecycleListener,"
"When writing a LifecycleListener that is managed by Guice, make sure to 
inject all dependencies lazily using {@link Provider} injection.  Otherwise,
these dependencies will be instantiated too early thereby bypassing lifecycle 
features in LifecycleModule."